browser.go

```diff
@@ -43,10 +43,22 @@ func (u *UserAgent) evalBrowserName(ua string) bool {
 		case strings.Contains(ua, "ucbrowser/") || strings.Contains(ua, "ucweb/"):
 			u.Browser.Name = BrowserUCBrowser
 
+		case strings.Contains(ua, "nintendobrowser/"):
+			u.Browser.Name = BrowserNintendo
+
+		case strings.Contains(ua, "samsungbrowser/"):
+			u.Browser.Name = BrowserSamsung
+
+		case strings.Contains(ua, "coc_coc_browser/"):
+			u.Browser.Name = BrowserCocCoc
+
 		// Edge, Silk and other chrome-identifying browsers must evaluate before chrome, unless we want to add more overhead
-		case strings.Contains(ua, "chrome/") || strings.Contains(ua, "crios/") || strings.Contains(ua, "chromium/") || strings.Contains(ua, "crmo/"):
+		case (strings.Contains(ua, "chrome/") || strings.Contains(ua, "crios/") || strings.Contains(ua, "chromium/") || strings.Contains(ua, "crmo/")) && !strings.Contains(ua, "yabrowser/"):
 			u.Browser.Name = BrowserChrome
 
+		case strings.Contains(ua, "yabrowser/"):
```

Comment:
You can move this check above the previous one and remove the `&& !strings.Contains(ua, "yabrowser/"`